static/src/javascripts/projects/journalism/jtbd-survey.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import ophan from 'ophan/ng';
+import { session as sessionStorage, local as localStorage } from 'lib/storage';
+import { campaignsFor } from 'common/modules/commercial/targeting-tool';
+import { init as initSurvey, BusinessError } from './survey';
+
+type Answer = number;
+type Question = number;
+type CampaignQuestion = {
+    question: string,
+    askWhy: boolean,
+};
+type CampaignFields = {
+    campaignId: string,
+    questions: CampaignQuestion[],
+};
+type Campaign = {
+    fields: CampaignFields,
+};
+
+// campaign settings: most of it will come from the targeting tool
+const campaignId: string = 'test-survey';
+// survey starts on Aug 23, 2017
+const startOfSurvey: Date = new Date('2017-07-23');
+const endOfSurvey: number = startOfSurvey.setMonth(
+    startOfSurvey.getMonth() + 1
+);
+
+/* Generate a [from .. to[ range */
+const range = (from: number, to: number): number[] => {
+    const rangerec = (n: number, acc: number[]): number[] => {
+        if (n >= to) {
+            return acc;
+        }
+        return rangerec(n + 1, acc.concat(n));
+    };
+    return rangerec(from, []);
+};
+
+/* Draw a random sample of 3 questions */
+const draw = (questions: Object[]): number[] => {
+    const lottery: number[] = range(0, questions.length);
+    let i;
+
+    i = Math.floor(Math.random() * lottery.length);
+    const q1 = lottery[i];
+
+    // ¯\_(ツ)_/¯
+    lottery.splice(i, 1);
+    i = Math.floor(Math.random() * lottery.length);
+    const q2 = lottery[i];
+
+    lottery.splice(i, 1);
+    i = Math.floor(Math.random() * lottery.length);
+    const q3 = lottery[i];
+
+    return [q1, q2, q3];
+};
+
+/* Choose a question out of the remaingin pool for this session */
+const selectQuestion = (qs: Question[], as: Answer[]): number => {
+    // indices of open questions
+    const oqs = as.reduce((acc, a, i) => (a === -1 ? acc.concat(i) : acc), []);
+
+    if (!oqs.length) {
+        return -1;
+    }
+
+    return oqs[Math.floor(Math.random() * oqs.length)];
+};
+
+/* Record answer and send data to Ophan */
+const save = (
+    qs: Question[],
+    as: Answer[],
+    q: Question,
+    answer: Answer,
+    why: ?string
+) => {
+    const hasSnippet: boolean = !!document.getElementsByClassName(
+        'explainer-snippet'
+    ).length;
+
+    as[q] = answer;
+    localStorage.set('gu.jtbd.answers', as, { expires: endOfSurvey });
+
+    ophan.record({
+        component: 'jtbd-survey',
+        hasSnippet,
+        qs,
+        as,
+        why,
```

Comment:
Ok I think we need to discuss more because the new ophan interaction model is directly intended to **not** *accept pretty much anything we stick in* as it has been abused and make usage of the data quite difficult for analysts

cc @oilnam who know more than me about the data lake   

I don't know as well enough about the `additional` event type but it seems it has not been designed to store interactions directly.